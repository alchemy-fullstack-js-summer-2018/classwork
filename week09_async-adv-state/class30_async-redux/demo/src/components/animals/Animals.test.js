import React from 'react';
import { shallow } from 'enzyme';
import Animals from './Animals';
import toJSON from 'enzyme-to-json';

describe('Animals', () => {
  
  it('renders add if no animal prop', () => {
    const animals = [
      { name: 'felix', type: 'cat', key: 'abc' },
      { name: 'garfield', type: 'cat', key: 'def' },
      { name: 'lassie', type: 'dog', key: 'ghi' },
    ];
    const wrapper = shallow(<Animals 
      animals={animals}
      onUpdate={jest.fn()}
      onRemove={jest.fn()}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});