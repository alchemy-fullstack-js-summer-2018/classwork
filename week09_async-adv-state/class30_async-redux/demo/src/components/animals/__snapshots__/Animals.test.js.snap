// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animals renders add if no animal prop 1`] = `
<ul>
  <Animal
    animal={
      Object {
        "key": "abc",
        "name": "felix",
        "type": "cat",
      }
    }
    key="abc"
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Animal
    animal={
      Object {
        "key": "def",
        "name": "garfield",
        "type": "cat",
      }
    }
    key="def"
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Animal
    animal={
      Object {
        "key": "ghi",
        "name": "lassie",
        "type": "dog",
      }
    }
    key="ghi"
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
</ul>
`;
